### PR TITLE
Add specific server response

### PR DESCRIPTION
**Pull request by chamil321**

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java [hunk 1]
```diff
@@ -62,26 +62,30 @@ public static void handleInbound(CarbonMessage cMsg, CarbonCallback callback) {
 
         String protocol = (String) cMsg.getProperty(org.wso2.carbon.messaging.Constants.PROTOCOL);
         if (protocol == null) {
+            cMsg.setProperty(org.wso2.carbon.transport.http.netty.common.Constants.HTTP_STATUS_CODE, 500);
```
Comment (hasithaa) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java]: Please find a alternative solution to set STATUS_CODE as this logic to applicable for Server connectors. Similar to #2458

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/parser/Node.java [hunk 2]
```diff
@@ -103,13 +103,18 @@ else if (matchLength == uriFragment.length()) {
...
+        try {
+            for (ResourceInfo resourceInfo : this.resource) {
+                if (resourceInfo.getAnnotationAttachmentInfo(Constants.HTTP_PACKAGE_PATH,
+                        requestDetails.get(Constants.HTTP_METHOD)) != null) {
+                    return resourceInfo;
+                }
             }
+            return tryMatchingToDefaultVerb(requestDetails);
+        } catch(Throwable e) {
+            requestDetails.put(Constants.HTTP_STATUS_CODE,"404");
```
Comment (shafreenAnfar) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/parser/Node.java]: Need to add a space after the ,

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/parser/Node.java [hunk 3]
```diff
@@ -103,13 +103,18 @@ else if (matchLength == uriFragment.length()) {
...
-                return resourceInfo;
+        try {
+            for (ResourceInfo resourceInfo : this.resource) {
+                if (resourceInfo.getAnnotationAttachmentInfo(Constants.HTTP_PACKAGE_PATH,
+                        requestDetails.get(Constants.HTTP_METHOD)) != null) {
+                    return resourceInfo;
+                }
             }
+            return tryMatchingToDefaultVerb(requestDetails);
+        } catch(Throwable e) {
```
Comment (chamil321) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/parser/Node.java]: yes, this.resource can be null. I experiences it when the request doesn't contains a resource.  Eg. http://localhost:9090/echo

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/ErrorHandlerUtils.java [hunk 4]
```diff
@@ -44,9 +44,9 @@ public static String getErrorMessage(Throwable throwable) {
         if (throwable instanceof StackOverflowError) {
             errorMsg = "fatal " + errorPrefix + "stack overflow ";
         } else if (throwable.getMessage() != null) {
-            errorMsg = errorPrefix + makeFirstLetterLowerCase(throwable.getMessage());
+            errorMsg = makeFirstLetterLowerCase(throwable.getMessage());
         } else {
-            errorMsg = errorPrefix;
+            errorMsg = "";
```
Comment (chamil321) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/ErrorHandlerUtils.java]: yes, Updated

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/http/HTTPErrorHandler.java [hunk 5]
```diff
@@ -42,7 +42,12 @@
 
     @Override
     public void handleError(Exception e, CarbonMessage carbonMessage, CarbonCallback callback) {
-        callback.done(createErrorMessage(e.getMessage(), 500));
+
+        Object carbonStatusCode = carbonMessage.getProperty
+                (org.wso2.carbon.transport.http.netty.common.Constants.HTTP_STATUS_CODE);
+        callback.done(createErrorMessage
+                (noEntityBodyCheck(e, carbonStatusCode), (carbonStatusCode == null) ? 500 :
```
Comment (chamil321) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/http/HTTPErrorHandler.java]: updated

File: modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/parser/Node.java [hunk 6]
```diff
@@ -103,13 +103,18 @@ else if (matchLength == uriFragment.length()) {
...
+            requestDetails.put(Constants.HTTP_STATUS_CODE,"404");
+            throw  new BallerinaException("Method :"+requestDetails.get(Constants.HTTP_METHOD));
```
Comment (chamil321) [inline modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/parser/Node.java]: updated

File: modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java (hunk 1 above)
Comment (chamil321) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java]: Solved the issue

**Pull request merged**